src/profile-logic/call-tree.js

```diff
@@ -496,42 +508,41 @@ function _getStackSelf(
     const callNodeIndex = sampleIndexToCallNodeIndex[sampleIndex];
     if (callNodeIndex !== null) {
       const weight = samples.weight ? samples.weight[sampleIndex] : 1;
-      callNodeSelf[callNodeIndex] += weight;
+      callNodeLeaf[callNodeIndex] += weight;
     }
   }
 
-  return { callNodeSelf, callNodeLeaf: callNodeSelf };
+  // Workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1858310
+  const abs = Math.abs;
+
+  let rootTotalSummary = 0;
+  for (let callNodeIndex = 0; callNodeIndex < callNodeCount; callNodeIndex++) {
+    rootTotalSummary += abs(callNodeLeaf[callNodeIndex]);
+  }
```

Comment:
I guess this could have stayed in the loop over the call nodes in computeCallTreeTimings like this was before? Now we're looping over call nodes twice :-)

Reply:
That's true. I did it this way so that rebasing #4900 would be easier.